machine/machine_core/machine.py

```diff
@@ -164,29 +165,33 @@ def journal_messages(self, matches, log_level, cursor=None):
         else:
             cursor_arg = ""
 
-        cmd = "journalctl 2>&1 %s -o cat -p %d %s || true" % (cursor_arg, log_level, " + ".join(matches))
-        messages = self.execute(cmd).splitlines()
+        cmd = "journalctl 2>&1 %s -o cat -p %d %s || true" % (cursor_arg, log_level, " + ".join(all_matches))
+        output = self.execute(cmd)
+        assert output, "journalctl did not return any output"
```

Comment:
Isn't the correct behaviour here rather to treat this as empty?  So like `execute(cmd) or ''`?  I'm not a big fan of `.execute()` returning `None` instead of the empty string.  Maybe we should actually fix that.

Reply:
I glanced over this comment, I'm all for changing m.execute to not return `None` this would be a quite big API breakage. From a quick scan in Cockpit I don't see anything relying on `m.execute` being able to return `None`.